src/charm.py

```diff
@@ -218,6 +219,46 @@ def _monitor_layer(self) -> Layer:
         }
         return Layer(layer_config)  # type: ignore
 
+    @property
+    def _log_rotate_layer(self) -> Layer:
+        """Returns a Pebble configuration layer for log rotate."""
+        container = self.unit.get_container(Config.CONTAINER_NAME)
+        # Before generating the log rotation layer we must first configure the log rotation
+        # template.
+        with open(Config.LogRotate.LOG_ROTATE_TEMPLATE, "r") as file:
+            template = jinja2.Template(file.read())
+
+        container.push(
+            Config.LogRotate.RENDERED_TEMPLATE,
+            template.render(
+                logs_directory=Config.LOG_DIR,
+                mongo_user=Config.UNIX_USER,
+                max_log_size=Config.LogRotate.MAX_LOG_SIZE,
+                max_rotations=Config.LogRotate.MAX_ROTATIONS_TO_KEEP,
+            ),
+            user=Config.UNIX_USER,
+            group=Config.UNIX_GROUP,
+        )
+
+        layer_config = {
+            "summary": "Log rotate layer",
+            "description": "Pebble config layer for rotating mongodb logs",
+            "services": {
+                "logrotate": {
+                    "summary": "log rotate",
+                    # Pebble errors out if the command exits too fast (1s).
+                    "command": f"sh -c 'logrotate {Config.LogRotate.RENDERED_TEMPLATE}; sleep 1'",
+                    "startup": "enabled",
+                    "override": "replace",
+                    "backoff-delay": "1m",
+                    "backoff-factor": 1,
+                    "user": Config.UNIX_USER,
+                    "group": Config.UNIX_GROUP,
+                }
+            },
+        }
+        return Layer(layer_config)  # type: ignore
```

Comment:
Why is there a need of `# type: ignore` here while we return what it advertised by the type hints?